File: solutions/3.py
from math import ceil

n = 600851475143

primeFactors = []

for i in range(2, ceil(n**0.5)):
    if n % i == 0:
        isPrime = True # default

        for p in primeFactors:
            if i % p == 0:
                isPrime = False

        if isPrime:
            primeFactors.append(i)

print(primeFactors[-1])

File: solutions/3.rb
n = 600851475143

prime_factors = []

(2...Math.sqrt(n).ceil).each do |i|
  if n % i == 0
    is_prime = true # default

    prime_factors.each do |p|
      is_prime = false if i % p == 0
    end

    prime_factors.push i if is_prime
  end
end

puts prime_factors[-1]
